tests/queries/0_stateless/02724_grace_hash_join.sql

```diff
@@ -0,0 +1,44 @@
+set join_algorithm = 'grace_hash';
```

Comment:
We can also reuse existing tests. Some tests are testing several algorithms and for grace_join there's check that it doesn't support `RIGHT` and `LEFT`. We can remove it and the result should be the same as for another algorithms.

https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/01721_join_implicit_cast_long.sql.j2#L12-L14

https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/02273_full_sort_join.sql.j2#L30-L32

https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/02274_full_sort_join_nodistinct.sql.j2#L18-L20

https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/02275_full_sort_join_long.sql.j2#L25-L29

Reply:
thinks very much for this.